apps/service_providers/llm_service/history_managers.py

```diff
@@ -123,13 +124,14 @@ def save_message_to_history(
         )
 
         if experiment_tag:
-            tag, _ = Tag.objects.get_or_create(
-                name=experiment_tag,
-                team=self.session.team,
-                is_system_tag=True,
-                category=TagCategories.BOT_RESPONSE,
-            )
-            chat_message.add_tag(tag, team=self.session.team, added_by=None)
+            with current_team(self.session.team):
```

Comment:
This should already be set at a higher level I think.

Also I just realized that if the current team IS already set then this will unset it after the context manager exits. Maybe we should update 'current_team' to reset back to what it was before.

Or else make sure that current team is set higher up the chain.

Reply:
Good points. I'll have a looksy